### YamlMime:ManagedReference
items:
- uid: System.NotFiniteNumberException
  id: NotFiniteNumberException
  children:
  - System.NotFiniteNumberException.#ctor
  - System.NotFiniteNumberException.#ctor(System.Double)
  - System.NotFiniteNumberException.#ctor(System.String)
  - System.NotFiniteNumberException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.NotFiniteNumberException.#ctor(System.String,System.Double)
  - System.NotFiniteNumberException.#ctor(System.String,System.Exception)
  - System.NotFiniteNumberException.#ctor(System.String,System.Double,System.Exception)
  - System.NotFiniteNumberException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.NotFiniteNumberException.OffendingNumber
  langs:
  - csharp
  name: NotFiniteNumberException
  nameWithType: NotFiniteNumberException
  fullName: System.NotFiniteNumberException
  type: Class
  summary: "The exception that is thrown when a floating-point value is positive infinity, negative infinity, or Not-a-Number (NaN)."
  remarks: "NotFiniteNumberException is available for programming languages that do not support the concepts of infinity and Not-a-Number in floating-point operations.       NotFiniteNumberException uses the HRESULT COR_E_NOTFINITENUMBER, which has the value 0x80131528.       For a list of initial property values for an instance of NotFiniteNumberException, see the <xref:System.NotFiniteNumberException.%23ctor%2A> constructors."
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class NotFiniteNumberException : ArithmeticException
  inheritance:
  - System.ArithmeticException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor
  id: '#ctor'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException()
  nameWithType: NotFiniteNumberException.NotFiniteNumberException()
  fullName: System.NotFiniteNumberException.NotFiniteNumberException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class."
  remarks: "This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as &quot;The number encountered was not a finite quantity.&quot; This message takes into account the current system culture.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.Exception.Message%2A>|The localized error message string.|"
  syntax:
    content: public NotFiniteNumberException ();
    parameters: []
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.Double)
  id: '#ctor(System.Double)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(Double)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(Double)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(Double)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with the invalid number."
  remarks: "The `offendingNumber` parameter must be both a <xref:System.Double> and an invalid number.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.NotFiniteNumberException.OffendingNumber%2A>|The invalid number.|"
  syntax:
    content: public NotFiniteNumberException (double offendingNumber);
    parameters:
    - id: offendingNumber
      type: System.Double
      description: "The value of the argument that caused the exception."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(String)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with a specified error message."
  remarks: "Ensure that the `message` string is understandable to the user and that it is localized for the current culture.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.Exception.Message%2A>|The error message string.|"
  syntax:
    content: public NotFiniteNumberException (string message);
    parameters:
    - id: message
      type: System.String
      description: "The message that describes the error."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(SerializationInfo,StreamingContext)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(SerializationInfo,StreamingContext)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with serialized data."
  remarks: "This constructor is called during deserialization to reconstitute the exception object transmitted over a stream."
  syntax:
    content: protected NotFiniteNumberException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "The object that holds the serialized object data."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "The contextual information about the source or destination."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Double)
  id: '#ctor(System.String,System.Double)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(String,Double)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Double)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Double)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with a specified error message and the invalid number."
  remarks: "Ensure that the `message` string is understandable to the user and that it is localized for the current culture.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|A null reference (`Nothing` in Visual Basic).|   |<xref:System.Exception.Message%2A>|The error message string.|   |<xref:System.NotFiniteNumberException.OffendingNumber%2A>|The invalid number.|"
  syntax:
    content: public NotFiniteNumberException (string message, double offendingNumber);
    parameters:
    - id: message
      type: System.String
      description: "The message that describes the error."
    - id: offendingNumber
      type: System.Double
      description: "The value of the argument that caused the exception."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(String,Exception)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Exception)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with a specified error message and a reference to the inner exception that is root cause of this exception."
  remarks: "An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property. The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|The inner exception reference.|   |<xref:System.Exception.Message%2A>|The error message string.|"
  syntax:
    content: public NotFiniteNumberException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "The error message that explains the reason for the exception."
    - id: innerException
      type: System.Exception
      description: "The exception that is the cause of the current exception. If the `innerException` parameter is not a null reference (<xref uid=&quot;langword_csharp_Nothing&quot; name=&quot;Nothing&quot; href=&quot;&quot;></xref> in Visual Basic), the current exception is raised in a <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref> block that handles the inner exception."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Double,System.Exception)
  id: '#ctor(System.String,System.Double,System.Exception)'
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: NotFiniteNumberException(String,Double,Exception)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Double,Exception)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Double,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: "Initializes a new instance of the <xref href=&quot;System.NotFiniteNumberException&quot;></xref> class with a specified error message, the invalid number, and a reference to the inner exception that is root cause of this exception."
  remarks: "An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property. The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.       The following table shows the initial property values for an instance of <xref:System.NotFiniteNumberException>.      |Property|Value|   |--------------|-----------|   |<xref:System.Exception.InnerException%2A>|The inner exception reference.|   |<xref:System.Exception.Message%2A>|The error message string.|   |<xref:System.NotFiniteNumberException.OffendingNumber%2A>|The invalid number.|"
  syntax:
    content: public NotFiniteNumberException (string message, double offendingNumber, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "The error message that explains the reason for the exception."
    - id: offendingNumber
      type: System.Double
      description: "The value of the argument that caused the exception."
    - id: innerException
      type: System.Exception
      description: "The exception that is the cause of the current exception. If the `innerException` parameter is not a null reference (<xref uid=&quot;langword_csharp_Nothing&quot; name=&quot;Nothing&quot; href=&quot;&quot;></xref> in Visual Basic), the current exception is raised in a <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref> block that handles the inner exception."
  overload: System.NotFiniteNumberException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.NotFiniteNumberException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: GetObjectData(SerializationInfo,StreamingContext)
  nameWithType: NotFiniteNumberException.GetObjectData(SerializationInfo,StreamingContext)
  fullName: System.NotFiniteNumberException.GetObjectData(SerializationInfo,StreamingContext)
  type: Method
  assemblies:
  - mscorlib
  namespace: System
  summary: "Sets the <xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;></xref> object with the invalid number and additional exception information."
  remarks: "<xref:System.TypeLoadException.GetObjectData%2A> sets a <xref:System.Runtime.Serialization.SerializationInfo> with all the exception object data targeted for serialization. During deserialization, the exception object is reconstituted from the <xref:System.Runtime.Serialization.SerializationInfo> transmitted over the stream.       For more information, see <xref:System.Runtime.Serialization.SerializationInfo>."
  syntax:
    content: public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "The object that holds the serialized object data."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "The contextual information about the source or destination."
  overload: System.NotFiniteNumberException.GetObjectData*
  exceptions:
  - type: System.ArgumentNullException
    commentId: T:System.ArgumentNullException
    description: "The <code>info</code> object is <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>."
  platform:
  - net462
- uid: System.NotFiniteNumberException.OffendingNumber
  id: OffendingNumber
  parent: System.NotFiniteNumberException
  langs:
  - csharp
  name: OffendingNumber
  nameWithType: NotFiniteNumberException.OffendingNumber
  fullName: System.NotFiniteNumberException.OffendingNumber
  type: Property
  assemblies:
  - mscorlib
  namespace: System
  summary: "Gets the invalid number that is a positive infinity, a negative infinity, or Not-a-Number (NaN)."
  remarks: "The OffendingNumber property returns the same value as was passed into the constructor, or zero if no value was supplied. Override the OffendingNumber property to customize the content or format of the invalid value.       This property is read-only."
  syntax:
    content: public double OffendingNumber { get; }
    return:
      type: System.Double
      description: "The invalid number."
  overload: System.NotFiniteNumberException.OffendingNumber*
  exceptions: []
  platform:
  - net462
references:
- uid: System.ArithmeticException
  isExternal: false
  name: System.ArithmeticException
- uid: System.ArgumentNullException
  isExternal: true
  name: System.ArgumentNullException
- uid: System.NotFiniteNumberException.#ctor
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException()
  nameWithType: NotFiniteNumberException.NotFiniteNumberException()
  fullName: System.NotFiniteNumberException.NotFiniteNumberException()
- uid: System.NotFiniteNumberException.#ctor(System.Double)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(Double)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(Double)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(Double)
- uid: System.Double
  parent: System
  isExternal: true
  name: Double
  nameWithType: Double
  fullName: System.Double
- uid: System.NotFiniteNumberException.#ctor(System.String)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(String)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.NotFiniteNumberException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(SerializationInfo,StreamingContext)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(SerializationInfo,StreamingContext)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Double)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(String,Double)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Double)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Double)
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Exception)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(String,Exception)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Exception)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.NotFiniteNumberException.#ctor(System.String,System.Double,System.Exception)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException(String,Double,Exception)
  nameWithType: NotFiniteNumberException.NotFiniteNumberException(String,Double,Exception)
  fullName: System.NotFiniteNumberException.NotFiniteNumberException(String,Double,Exception)
- uid: System.NotFiniteNumberException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.NotFiniteNumberException
  isExternal: false
  name: GetObjectData(SerializationInfo,StreamingContext)
  nameWithType: NotFiniteNumberException.GetObjectData(SerializationInfo,StreamingContext)
  fullName: System.NotFiniteNumberException.GetObjectData(SerializationInfo,StreamingContext)
- uid: System.NotFiniteNumberException.OffendingNumber
  parent: System.NotFiniteNumberException
  isExternal: false
  name: OffendingNumber
  nameWithType: NotFiniteNumberException.OffendingNumber
  fullName: System.NotFiniteNumberException.OffendingNumber
- uid: System.NotFiniteNumberException.#ctor*
  parent: System.NotFiniteNumberException
  isExternal: false
  name: NotFiniteNumberException
  nameWithType: NotFiniteNumberException.NotFiniteNumberException
- uid: System.NotFiniteNumberException.GetObjectData*
  parent: System.NotFiniteNumberException
  isExternal: false
  name: GetObjectData
  nameWithType: NotFiniteNumberException.GetObjectData
- uid: System.NotFiniteNumberException.OffendingNumber*
  parent: System.NotFiniteNumberException
  isExternal: false
  name: OffendingNumber
  nameWithType: NotFiniteNumberException.OffendingNumber
